packages/providers/src.ts/base-provider.ts

```diff
@@ -808,11 +817,38 @@ export class BaseProvider extends Provider {
         // }
     }
 
-    async getLogs(filter: Filter | FilterByBlockHash | Promise<Filter | FilterByBlockHash>): Promise<Array<Log>> {
-        await this.getNetwork();
+    async getLogs(filter: Filter | Promise<Filter>): Promise<Array<Log>> {
+        if (!this._mirrorNodeUrl) logger.throwError("missing provider", Logger.errors.UNSUPPORTED_OPERATION);
```

Comment:
Let's move this into a separate function in order to avoid code duplication. (mirror node url check)